### PR TITLE
compare two dict

### PR DESCRIPTION
**Pull request by Oscer2016**
implement a method to compare two dictionaries

if dict_1 contains dict_2, return 1.

if dict_2 contains dict_1, return -1.

otherwise return 2

Signed-off-by: He Pan <hepan#xiyoulinux.org> #to@

### CODE REVIEW

File: app/tests/utils/dictCmpTests.py [hunk 1]
```diff
@@ -18,9 +18,85 @@ def test_two_flat_dicts_equal(self):
             "project": "lepv"
         }
 
+        dict_3 = {}
+
+        dict_4 = {}
+
+        comp_result_1 = DictUtil.compare(dict_1, dict_2)
+        comp_result_2 = DictUtil.compare(dict_3, dict_4)
```
Comment (macxu) [inline app/tests/utils/dictCmpTests.py]: we should have separate test case for this, say "two_empty_dicts_should_equal".

File: app/tests/utils/dictCmpTests.py [hunk 2]
```diff
@@ -18,9 +18,85 @@ def test_two_flat_dicts_equal(self):
...
+        }
+
+        dict_4 = {
+            "project1": {"k1": "v1", "k2": "v2"},
+            "project2": "lepv2"
+        }
+
+        comp_result_1 = DictUtil.compare(dict_2, dict_1)
+        comp_result_2 = DictUtil.compare(dict_4, dict_3)
+        self.assertEqual(comp_result_1, 1, 'the comparison result of two identical flat dicts should be 1')
```
Comment (macxu) [inline app/tests/utils/dictCmpTests.py]: 1. correct the error messages.
2. create separate test cases for it, and give it a meaningful name to clarify the test point.

File: app/tests/utils/dictCmpTests.py (hunk 1 above)
Comment (Oscer2016) [reply app/tests/utils/dictCmpTests.py]: ok

File: app/tests/utils/dictCmpTests.py [hunk 3]
```diff
@@ -18,9 +18,85 @@ def test_two_flat_dicts_equal(self):
...
+        self.assertEqual(comp_result_2, 1, 'the comparison result of two identical flat dicts should be 1')
+
+    def test_two_flat_dicts_contained(self):
```
Comment (macxu) [inline app/tests/utils/dictCmpTests.py]: the test ( the method name ) indicates this is about "flat" dicts but this is not true.

File: app/tests/utils/dictCmpTests.py [hunk 4]
```diff
@@ -18,9 +18,85 @@ def test_two_flat_dicts_equal(self):
...
+        dict_2 = {
+            "project1": {"k1": "v1", "k2": "v2"},
+            "project2": "lepv2"
+        }
+
         comp_result = DictUtil.compare(dict_1, dict_2)
-        self.assertEqual(comp_result, 0, 'the comparison result of two identical flat dicts should be 0')
+        self.assertEqual(comp_result, -1, 'the comparison result of two identical flat dicts should be -1')
+
+    def test_two_flat_dicts_other(self):
```
Comment (macxu) [inline app/tests/utils/dictCmpTests.py]: same here:
1. is the test name accurate?
2. a test case should be highly focused, validate one thing only.

**Pull request merged**